packages/contentstack-auth/src/utils/mfa-handler.ts

```
@@ -0,0 +1,147 @@
+        );
+      } else {
+        secret = envSecret;
+        source = 'environment variable';
+      }
+    }
+
+    if (!secret) {
+      log.debug('Checking stored MFA secret', { module: 'mfa-handler' });
+      const mfaConfig = configHandler.get('mfa');
+      if (mfaConfig?.secret) {
+        try {
+          secret = this.encrypter.decrypt(mfaConfig.secret);
+          source = 'stored configuration';
+        } catch (error) {
+          log.debug('Failed to decrypt stored MFA secret', { module: 'mfa-handler', error });
+          cliux.print('CLI_AUTH_MFA_DECRYPT_FAILED', { color: 'yellow' });
+          handleAndLogError(error, { module: 'mfa-handler' }, messageHandler.parse('CLI_AUTH_MFA_DECRYPT_FAILED'));
+        }
+      }
+    }
+
+    if (secret) {
+      try {
+        const code = this.generateMFACode(secret);
+        log.debug('Generated MFA code', { module: 'mfa-handler', source });
+        return code;
+      } catch (error) {
+        log.debug('Failed to generate MFA code', { module: 'mfa-handler', error, source });
+        cliux.print('CLI_AUTH_MFA_GENERATION_FAILED', { color: 'yellow' });
+        cliux.print('CLI_AUTH_MFA_RECONFIGURE_HINT');
+        cliux.print(
+          'For more information about MFA, visit: https://www.contentstack.com/docs/developers/security/multi-factor-authentication',
+          { color: 'yellow' },
+        );
+      }
+    }
+  }
+
+  /**
+   * Gets MFA code through manual user input
+   * @returns Promise<string> The MFA code
+   * @throws Error if code format is invalid
+   */
+  async getManualMFACode(): Promise<string> {
+    const code = await askOTP();
+    if (!/^\d{6}$/.test(code)) {
+      log.debug('Invalid MFA code format', { module: 'mfa-handler', code });
+      cliux.print('CLI_AUTH_MFA_INVALID_CODE', { color: 'yellow' });
+      handleAndLogError(new Error(messageHandler.parse('CLI_AUTH_MFA_INVALID_CODE')), { module: 'mfa-handler' });
+    }
+    return code;
```

@cs-raj , it should be in try catch block and throw the error instead of logging (handleAndLogError).

---

Done